assignments/week-6/assign7.html

```diff
@@ -0,0 +1,84 @@
+<html> 
+<body> 
+
+<scrpit> 
+
+	class Table { 
+		constructor(id, size) { 
+			this.id = id; 
+			this.size = size; 
+			this.isOccupied = false; 
+			this.occupiedUntil = 0; 
+		} 
+	
+		occupy(time, partySize) { 
+			this.isOccupied = true; 
+			this.occupiedUntil = time + 	Math.floor(Math.random () * 3) + 1 
+			console.log('Table ${this.id} (size ${this.size})is now occupied by party of ${partySize} unitl hour ${this.occupiedUntil}.'); 
+	} 
+
+		free(time) { 
+			if (this.isOccupied && time => this.occupiedUntil { 
+				this.isOccupied = false; 
+				console.log('Table ${this.id} (size ${this.size}) is now free') 
+			}
+		} 
+	}
+	
+	class Resturant {
+		constrocutor(tables) { 
+			this.tables = tables; 
+		}
+		
+		seatParty(partySize, currentTime) { 
+	
+		const availableTable = this.tables
+			.filter(table => !tabe.isOccupied && table.size >= partySize) 
+			.sort ((a, b) =. a.size - b.size)[0]; 
+			
+		if (availabletable) {
+			availableTable.occupy(partySize, currentTime); 
+		} else {
+            console.log(`Party of ${partySize} is angry and leaves at hour ${currentTime}. No available table.`);
```

Comment:
nit: this method should return some sort of status code (or maybe throw an exception) to indicate that no tables are available, and then the caller can decide what to do with that status in terms of presenting to the user.